crescent/plugin.py

```diff
@@ -215,7 +223,16 @@ def __init__(
         self._unload_hooks: list[PluginCallbackT] = []
 
     def include(self, obj: T) -> T:
-        add_hooks(self, obj)
+        if isinstance(obj.metadata, AppCommandMeta):
+            obj.metadata.add_hooks(self.command_hooks, after=False)
+            obj.metadata.add_hooks(self.command_after_hooks, after=True)
+        if isinstance(obj.metadata, EventMeta):
+            obj.metadata.add_hooks(  # pyright: ignore [reportUnknownMemberType]
+                self.event_hooks, after=False
+            )
+            obj.metadata.add_hooks(  # pyright: ignore [reportUnknownMemberType]
+                self.event_after_hooks, after=True
+            )
```

Comment:
I feel like I've seen this code block more than once, should it be a function maybe?

Reply:
I think so